File: src/main/java/ru/trukhmanov/twochairsbackend/entity/Deck.java
package ru.trukhmanov.twochairsbackend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "decks")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class Deck {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="type", nullable=false, length=16)
    private String type; // DEFAULT/PAID/USER

    @Column(name="visibility", nullable=false, length=16)
    private String visibility; // PRIVATE/PUBLIC

    @Column(name="title", nullable=false, length=80)
    private String title;

    @Column(name="description")
    private String description;

    @Column(name="age_rating", nullable=false)
    private int ageRating;

    @Column(name="price_rub", nullable=false)
    private int priceRub;

    @Column(name="owner_user_id")
    private Long ownerUserId;

    @Column(name="is_published", nullable=false)
    private boolean published;
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/entity/Deck.kt
package ru.trukhmanov.twochairsbackend.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "decks")
class Deck(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "type", nullable = false, length = 16)
    var type: String = "",

    @Column(name = "visibility", nullable = false, length = 16)
    var visibility: String = "",

    @Column(name = "title", nullable = false, length = 80)
    var title: String = "",

    @Column(name = "description")
    var description: String? = null,

    @Column(name = "age_rating", nullable = false)
    var ageRating: Int = 0,

    @Column(name = "price_rub", nullable = false)
    var priceRub: Int = 0,

    @Column(name = "owner_user_id")
    var ownerUserId: Long? = null,

    @Column(name = "is_published", nullable = false)
    var published: Boolean = false
)
